_posts/2021-09-15-drain-cleaner.md

```diff
@@ -0,0 +1,190 @@
+    "value": {
+      "podDisruptionBudget": {
+        "maxUnavailable": 0
+      }
+    }
+  },
+  {
+    "op": "add",
+    "path": "/spec/kafka/template",
+    "value": {
+      "podDisruptionBudget": {
+        "maxUnavailable": 0
+      }
+    }
+  },
+]'
+```
+
+## Validating webhook configuration
+
+To register the Drain Cleaner's endpoint at the Kubernetes control-plane level, you need to create a 
+ValidatingWebhookConfiguration resource. This is a simple REST endpoint, exposed under `/drainer` URL path. When an 
+incoming API request matches one of the specified operations, groups, versions, resources, and scope for any of defined 
+rules, the request is sent to that endpoint.
+
+```yaml
+    rules:
+      - apiGroups:   [""]
+        apiVersions: ["v1"]
+        operations:  ["CREATE"]
+        resources:   ["pods/eviction"]
+        scope:       "Namespaced"
+    clientConfig:
+      service:
+        namespace: "strimzi-drain-cleaner"
+        name: "strimzi-drain-cleaner"
+        path: /drainer
+        port: 443
+      caBundle: $CA_BUNDLE
+```
+
+Here I used a placeholder for the PEM encoded CA bundle, which is required to validate the webhook's certificate. You 
+can use a self-signed CA certificate, but the Drain Cleaner's end-entity certificate, signed by that CA, must have a 
+common name (CN) that matches the application's service name.
+
+Let's see how that CA bundle looks like for a self-signed example deployed on a test namespace (in the git repository, 
+you can find detailed instructions on how to generate these certificates or let the platform do it for you):
+
+```sh
+# Self-signed CA certificate  
+Certificate:
+    Signature Algorithm: sha256WithRSAEncryption
+        Issuer: C=IT, ST=Rome, O=Strimzi, CN=drain-cleaner-ca.strimzi.io
+        Subject: C=IT, ST=Rome, O=Strimzi, CN=drain-cleaner-ca.strimzi.io
+        X509v3 extensions:
+            X509v3 Basic Constraints: critical
+                CA:TRUE, pathlen:0
+
+# Drain Cleaner's end-entity certificate         
+Certificate:
+    Signature Algorithm: sha1WithRSAEncryption
+        Issuer: C=IT, ST=Rome, O=Strimzi, CN=drain-cleaner-ca.strimzi.io
+        Subject: C=IT, ST=Rome, O=Strimzi, CN=drain-cleaner-webhook.strimzi.io
+        X509v3 extensions:
+            X509v3 Basic Constraints: critical
+                CA:FALSE
+            X509v3 Key Usage: 
+                Digital Signature, Key Encipherment
+            X509v3 Extended Key Usage: 
+                TLS Web Server Authentication, TLS Web Client Authentication
+            X509v3 Subject Alternative Name: 
+                DNS:strimzi-drain-cleaner, 
+                DNS:strimzi-drain-cleaner.test, 
+                DNS:strimzi-drain-cleaner.test.svc, 
+                DNS:strimzi-drain-cleaner.test.svc.cluster.local
+```
+
+## Let the magic happen
+
+The Drain Cleaner must be deployed at the Kubernetes cluster level using a cluster-admin user, as we need to create a 
```

Comment:
I don't like saying that it needs to be deployed with cluster-admin user. It can be deployed by any user who can create the resources => mainly the ValidatingWebhook. But that is not cluster admin.